docs/stencil-docs/deploying-a-theme/theme-best-practices.md

```diff
@@ -0,0 +1,169 @@
+```
+The WebDAV import option allows you to upload images in bulk. However, we suggest that you select the specific files needed and not import the entire markup file or unessential files as this can affect page load time.
+
+To learn about the BigCommerce folder structure best practices, see [Folder Structure](https://support.bigcommerce.com/s/article/File-Access-WebDAV#folder).
+
+The second way to import images is to use the image manager within the control panel.
+1. In the control panel, go to **Storefront** > **Image Manager**.
+2. Click **Upload images...**
+3. Click **Choose File** (or Browse) and select an image from your computer. Repeat for each additional image.
+4. Click **Upload**.
+
+When importing images, ensure images are appropriately-sized and have low byte size to increase a store's site speed. We recommend using a recent Stencil theme that uses responsive images. Once you have that, the best practice is to upload the best quality images you have, and BigCommerce will optimize them. We recommend JPEGs over other formats. See [Optimizing your Images](https://support.bigcommerce.com/s/article/Optimizing-Your-Images) for more details on how to optimize images.
+
+## Injecting variables
+
+Injecting javascript context variables allow you to access store data through your theme. You can inject javascript context variables using the `{{inject}}` helper for properties you want responses for by assigning a custom variable name. You can then use the `{{jsContext}}` handlebars helper to access a stringified JSON object containing all injected data with your assigned variable name as the key in key-value pairs. For more information, see [injection helpers](https://developer.bigcommerce.com/stencil-docs/reference-docs/handlebars-helpers-reference#injection-helpers). The code example below uses `{{inject}}` and `{{jsContent}}` to log the product name to your browser's console when added to Cornerstone's `product.html` file: 
+
+```javascript
+{{inject "myProductName" product.title}}
+
+<script>
+// Note the lack of quotes around the jsContext handlebars helper, it becomes a string automatically.
+var jsContext = JSON.parse({{jsContext}});
+/* jsContext would output "{\"myProductName\": \"Sample Product\"}" which can feed directly into
+your JavaScript. */
+console.log(jsContext.myProductName); // Will output: Sample Product
+</script>
+```
+
+We suggest injecting only specific variables and only the variables you need. If you inject all settings, you could accidently inject PII (Personally-Identifying Information) and cause performance issues with your site.
+
+## Using Lighthouse
+
+Google [Lighthouse](https://developers.google.com/web/tools/lighthouse) is an open-source tool used to improve the performance, quality, accessibility, best practices, and more for web pages. Enter a URL into Lighthouse, and it will generate a series of audits against the page. We highly encourage you to use Lighthouse as a validation step.
```

Comment:
```suggestion
Google [Lighthouse](https://developers.google.com/web/tools/lighthouse) is an open-source tool used to improve web page performance, quality, and accessibility. Enter a URL into Lighthouse, and it will generate a series of audits against the page. We highly encourage you to use Lighthouse as a validation step.
```